Sources/Memberlist.php

```diff
@@ -426,7 +426,7 @@ function MLSearch()
 		// Build the column link / sort information.
 		foreach ($context['columns'] as $col => $column_details)
 		{
-			$context['columns'][$col]['href'] = $scripturl . '?action=mlist;sa=search;start=0;sort=' . $col;
+			$context['columns'][$col]['href'] = $scripturl . '?action=mlist;sa=search;start=' . $_REQUEST['start'] . ';sort=' . $col;
```

Comment:
It looks like `$_REQUEST['start']` has not yet been sanitized at this point in `MLSearch()`. It should probably be cast to an integer before being used here.